code/06-building-decays/build_decays.py

```diff
@@ -0,0 +1,39 @@
+"""Build D* -> D0 (->K pi) pi decays from scratch."""
+
+from CommonParticles.StdAllNoPIDsPions import StdAllNoPIDsPions as Pions
+from CommonParticles.StdAllLooseKaons import StdAllLooseKaons as Kaons
+
+d0_daughters = {'pi-': '(PT > 750*MeV) & (P > 4000*MeV) & (MIPCHI2DV(PRIMARY) > 4)',
+                'K+': '(PT > 750*MeV) & (P > 4000*MeV) & (MIPCHI2DV(PRIMARY) > 4)'}
+d0_comb = "(AMAXDOCA('') < 0.2*mm) & (ADAMASS('D0') < 100*MeV)"
+d0_mother = "(VFASPF(VCHI2/VDOF)< 9) & (BPVDIRA > 0.9997) & (ADMASS('D0') < 70*MeV)"
+
+from Configurables import CombineParticles
+d0 = CombineParticles("Combine_D0",
+                      Decay='([D0 -> pi- K+]CC)',
+                      DaughtersCuts=d0_daughters,
+                      CombinationCut=d0_comb,
+                      MotherCut=d0_mother)
+
+from PhysSelPython.Wrappers import Selection
+d0_sel = Selection("Sel_D0",
+                   Algorithm=d0,
+                   RequiredSelections=[Pions, Kaons])
+
+dstar = CombineParticles("CombineDstar",
+                         Decay='[D*(2010)+ -> D0 pi+]cc',
+                         DaughtersCuts={'pi+': '(TRCHI2DOF < 3) & (PT > 100*MeV)'}
+                         CombinationCut="(ADAMASS('D*(2010)+') < 400*MeV)",
+                         MotherCut="(abs(M-MAXTREE('D0'==ABSID,M)-145.42) < 10*MeV) & (VFASPF(VCHI2/VDOF)< 9)")
+dstar_sel = Selection("Sel_Dstar",
+                      Algorithm=dstar,
+                      RequiredSelections=[d0_sel, Pions])
+
+from PhysSelPython.Wrappers import SelectionSequence
+dstar_seq = SelectionSequence('Dstar_Seq', TopSelection=dstar_sel)
+
+from Configurables import DaVinci
+DaVinci().UserAlgorithms += [dstar_seq.sequence()]
+
+
+# EOF
```

Comment:
To make this more consistent with (most) of [the other files](https://github.com/lhcb/first-analysis-steps/tree/1400d6a64fcd697a37d3b99e66887dab1e6cc172/code}):

1. Move all imports to the top of the file, combining all the imports from `Configurables`;
2. All double quotes `"` to single `'`;
3. Remove last three lines

Seeing a the cuts for the D0 `CombineParticles` are made as variables like `d0_daughters`, should the same be done for the D*+ `CombineParticles`?

Personally, I would not alias the `Std*` containers. I think it makes the `Selection` instantiations less obvious.